### DISCUSSION TITLE
Auto-Create - Cascade Create

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#13405] [Feature Requests]
Given the scenario: I want to have each directus_user having a "profile"

Case 1: I can now let the directus_user keep the relation to "profile"
or
Case 2: having the "profile" keep the relation to the directus_user

In either way i will have to create a hook to to create the profile. I would be a nice thing, to have a profile automaticly created when a user is created therefore the name "Cascade Create".
When setting a relationship the is a field "Default Value", but i could not find in the docs if this value to solve Case 1.

**Comment by rijkvanzanten** [comment]
You could also create the user with an empty object as the `profile` to have the API auto-create the nested profile (but without any data) 🤔 You could then also add this to the "Field Presets" of the role that creates users to make sure that "profile" field is always initialized with an empty object 🤔

**Comment by NilsBaumgartner1994** [comment]
I have done that. I solved by this, that a user can log in via different sso and still has access to his profile (after linking), or that a device is always created where I save additional informations which don’t belong directly to the directus_user


Beste Grüße,
Nils Baumgartner

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
